packages/client/src/views/CorrectionForm/CorrectionSummary.tsx

```diff
@@ -145,9 +145,13 @@ class CorrectionSummaryComponent extends React.Component<IFullProps, IState> {
         this.props.offlineResources.config.CURRENCY &&
         this.props.offlineResources.config.CURRENCY['isoCode']
     })[0].symbol
+
     ;(
       this.group.fields[0].nestedFields as any
-    ).REQUIRED[0].label.defaultMessage += ` ${currency}`
+    ).REQUIRED[0].label.defaultMessage = this.props.intl.formatMessage(
+      messages.correctionSummaryTotalPaymentLabel,
+      { currency }
+    )
```

Comment:
Where is this label being used? Could the internationalisation bit be moved into the rendering logic? Mutating the form field object in the `componentDidMount` method to make the currency label render correctly seems a bit like a code smell.

No need to change anything now as this component seems massive and very complex. More of a question for the future I guess :)

Reply:
> Where is this label being used? Could the internationalisation bit be moved into the rendering logic? Mutating the form field object in the `componentDidMount` method to make the currency label render correctly seems a bit like a code smell.
> 
> No need to change anything now as this component seems massive and very complex. More of a question for the future I guess :)

This label is being used in fees collection part of correction summary page, is defined in payment.ts file` nestedFields: {
        REQUIRED: [
          {
            name: 'totalFees',
            type: 'NUMBER',
            label: {
              defaultMessage: '',
              id: 'form.field.label.totalFees',
              description: 'Label for input Reason for Change'
            },
            required: true,
            initialValue: '',
            validate: [],
            mapping: {}
          },
          {
            name: 'proofOfPayment',
            type: 'SIMPLE_DOCUMENT_UPLOADER',
            label: messages.correctionSummaryProofOfPayment,
            description: messages.correctionSummaryProofOfPayment,
            required: true,
            initialValue: '',
            validate: [],
            mapping: {}
          }
        ]
      }`